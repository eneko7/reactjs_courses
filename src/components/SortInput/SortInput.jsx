/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
import React, { Fragment, Component } from 'react';
import propTypes from 'prop-types';
import styles from './SortInput.scss';

class SortInput extends Component {
  changeValue = (el) => {
    const { value, name } = el.target;
    const { updateFilterSearch, updateFilterSort, updateSearchRequest } = this.props;
    if (name === 'searchBy') {
      updateFilterSearch(value);
      updateSearchRequest('');
    } else {
      const { searchBy, searchRequest } = this.props;
      updateFilterSort(searchRequest, searchBy, value);
    }
  };

  render() {
    const {
      name,
      text,
      id,
      inputClass,
      labelClass,
      defaultChecked,
    } = this.props;
    return (
      <Fragment>
        <input type="radio" name={name} id={id} value={id} className={styles[inputClass]} defaultChecked={defaultChecked} onChange={this.changeValue} />
        <label htmlFor={id} className={styles[labelClass]}>{text}</label>
      </Fragment>
    );
  }
}

SortInput.defaultProps = {
  defaultChecked: false,
  searchRequest: '',
};

SortInput.propTypes = {
  name: propTypes.string.isRequired,
  text: propTypes.string.isRequired,
  id: propTypes.string.isRequired,
  inputClass: propTypes.string.isRequired,
  labelClass: propTypes.string.isRequired,
  searchBy: propTypes.string.isRequired,
  searchRequest: propTypes.string,
  defaultChecked: propTypes.bool,
  updateFilterSearch: propTypes.func.isRequired,
  updateFilterSort: propTypes.func.isRequired,
  updateSearchRequest: propTypes.func.isRequired,
};

export default SortInput;
