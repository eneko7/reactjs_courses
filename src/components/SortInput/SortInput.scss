@import "../../assets/constants/constants.scss";

.sort_by_button {
  cursor: pointer;
  font-size: 18px;
  line-height: 25px;
  font-family: $fontFamilyBold;
}

.sort_by_input {
  display: none;
}

.sort_by_input:checked + .sort_by_button {
  color: $roseColor;
}

.search_type_button {
  cursor: pointer;
  background-color: #343939;
  color: #fff;
  border: none;
  padding: 5px 30px;
  font-size: 16px;
  text-transform: uppercase;
  font-family: $fontFamilyBold;
}

.search_type_input {
  display: none;
}

.search_type_input:checked + .search_type_button {
  background-color: $roseColor;
}

@media (max-width: $smallPhone) {
  .search_type_button {
    padding: 5px 23px;
  }
}
