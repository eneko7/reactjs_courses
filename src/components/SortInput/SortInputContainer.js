import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import SortInput from './SortInput';
import { updateFilterSearch, updateFilterSort, updateSearchRequest } from '../../modules/filter/filterActions';
import { fetchSearchMovies } from '../../modules/movies/moviesActions';
import { getSearchBy, getSortBy, getSearchRequest } from '../../modules/filter/filterSelectors';

const mapStateToProps = state => ({
  searchBy: getSearchBy(state),
  sortBy: getSortBy(state),
  searchRequest: getSearchRequest(state),
});

const mapDispatchToProps = {
  updateFilterSearch,
  updateSearchRequest,
  updateFilterSort,
  fetchSearchMovies,
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SortInput));
