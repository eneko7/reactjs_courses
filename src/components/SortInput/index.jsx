import SortInput from './SortInputContainer';

export default SortInput;
