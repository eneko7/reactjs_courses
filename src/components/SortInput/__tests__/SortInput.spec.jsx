import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SortInput from '../SortInput';

const renderer = new ShallowRenderer();
const data = {
  name: 'name',
  text: 'text',
  id: 'id',
  inputClass: 'inputClass',
  labelClass: 'labelClass',
  defaultChecked: 'defaultChecked',
};
renderer.render(<SortInput {...data} />);
describe('SortInput', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
