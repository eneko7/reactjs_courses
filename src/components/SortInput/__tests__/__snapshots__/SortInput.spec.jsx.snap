// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortInput renders correctly 1`] = `
<React.Fragment>
  <input
    className="inputClass"
    defaultChecked="defaultChecked"
    id="id"
    name="name"
    type="radio"
  />
  <label
    className="labelClass"
    htmlFor="id"
  >
    text
  </label>
</React.Fragment>
`;
